### PR TITLE
Introduce Kustomize and Tilt

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Introduce Kustomize manifests in `testing/infra/k8s`, using ECK for running Elasticsearch, Kibana, Fleet Server, and an Elastic Agent with the APM integration injected.

We use the Kustomize manifests from Tiltfile, injecting a live-reloaded custom image into the apm-server Agent pod spec.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Install Tilt: https://docs.tilt.dev/install.html. On macOS you can use Docker Desktop, on Linux you should use Kind (optionally using https://github.com/tilt-dev/ctlptl for managing the cluster.)
2. Run `tilt up`
3. Send some APM data, check that it shows up in Kibana

## Related issues

None

### CODE REVIEW

**Review by simitt — APPROVED** [review]
Works great 🚀 . The one thing missing is to add a small documentation section. It might be worth adding something directly to the README. 

Minor note: maybe also mention `tilt down` for how to stopp the deployment again.

**Comment by axw** [comment]
@simitt I've just updated `dev_docs/TESTING.md` for now. I think it might be worth updating README.md too, but would rather take my time on that and don't want to delay merging this.

File: Tiltfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#
+# Without this, ECK refuses to recreate the trial license. That's expected
+# for actual trial use, but we rely an Enterprise feature for local development:
+# installing the integration package by uploading to Fleet.
+if config.tilt_subcommand == "down":
+  print(local("kubectl delete --ignore-not-found namespace/elastic-system"))
+
+# Add a button for sending trace events and metrics to APM Server.
+load('ext://uibutton', 'cmd_button')
+cmd_button(
```
Comment (marclop) [inline Tiltfile:41]: This is pretty cool

File: testing/infra/k8s/base/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+resources:
+  - https://download.elastic.co/downloads/eck/2.4.0/crds.yaml
+  - https://download.elastic.co/downloads/eck/2.4.0/operator.yaml
```
Comment (marclop) [inline testing/infra/k8s/base/kustomization.yaml:3]: I wonder how we'll keep these up to date.

**Review by marclop — COMMENTED** [review]
excited to try this out!

File: testing/infra/k8s/base/kustomization.yaml (hunk 2 above)
Comment (axw) [reply testing/infra/k8s/base/kustomization.yaml:3]: Like end users, I don't think it's too much of a problem if we're not always on the latest version. Only if there's some security issue or critical bug, or a new feature that we want to take advantage of.

**Pull request merged**